File: Dante1/Module1.07/Zadanie7.19/main.c
#include <stdio.h>

int is_prime(int a)
{
    int counter = 0;
    if(a == 0 || a == 1 || a < 0)
    {
        return 0;
    }
    for(int i = 1; i <= a; i++)
    {
        if(a % i == 0)
        {
            counter++;
        }
    }
    if(counter <= 2)
    {
        return 1;
    }
    return 0;
}


int main()
{
    int matrix[5][5];

    printf("Enter:\n");
    for(int i = 0; i < 5; i++)
    {
        for(int j = 0; j < 5; j++)
        {
            int error = scanf("%i", &matrix[i][j]);
            if(error != 1)
            {
                printf("Incorrect input");
                return 1;
            }
        }
    }

    int max_primes = 0;
    int count = 0;

    for (int i = 0; i <= 2; i++)
    {
        for (int j = 0; j <= 2; j++)
        {
            int prime_count = 0;

            for (int x = 0; x < 3; x++)
            {
                for (int y = 0; y < 3; y++)
                {
                    if (is_prime(matrix[i + x][j + y]))
                    {
                        prime_count++;
                    }
                }
            }

            if (prime_count > max_primes)
            {
                max_primes = prime_count;
                count = 1;
            }
            else if (prime_count == max_primes)
            {
                count++;
            }
        }
    }

    if (max_primes == 0)
    {
        printf("0");
    }
    else
    {
        printf("%d", count);
        for (int i = 0; i <= 2; i++)
        {
            for (int j = 0; j <= 2; j++)
            {
                int prime_count = 0;

                for (int x = 0; x < 3; x++)
                {
                    for (int y = 0; y < 3; y++)
                    {
                        if (is_prime(matrix[i + x][j + y]))
                        {
                            prime_count++;
                        }
                    }
                }

                if (prime_count == max_primes)
                {
                    printf("\n%i %i", i + 1, j + 1);
                }
            }
        }
    }

    return 0;
}

File: Dante1/Module1.07/Zadanie7.19/main.rs
fn is_prime(a: i32) -> i32  {
    let mut counter: usize = 0;
    if a == 0 || a == 1 || a < 0 {
        return 0;
    }

    for i in 1..=i32::isqrt(a) {
        if a % i == 0 {
            counter += 1;
        }
    }
    if counter <= 2 {
        return 1;
    }

    0
}

fn basic_cycle(mut matrix: &[[i32; 5]; 5], mut max_primes: usize, mut counter: usize) {
    for i in 0..=2 {
        for j in 0..=2 {
            let mut helper: usize = 0;

            for x in 0..3 {
                for y in 0..3 {
                    if is_prime(matrix[i + x][j + y]) == 1
                    {
                        helper += 1;
                    }
                }
            }

            if helper > max_primes {
                max_primes = helper;
                counter = 1;
            }
            else if max_primes == helper {
                counter += 1;
            }
        }
    }
}

fn cool_cycle(mut matrix: &[[i32; 5]; 5], mut max_primes: usize, mut counter: usize) {
    for i in 0..=2 {
        for j in 0..=2 {
            let mut helper: usize = 0;

            for x in 0..3 {
                for y in 0..3 {
                    if is_prime(matrix[i + x][j + y]) == 1
                    {
                        helper += 1;
                    }
                }
            }

            if max_primes == helper {
                println!("{} {}", i + 1, j + 1);
            }
        }
    }
}

fn main() {
    let mut input = String::new();

    let mut matrix: [[i32; 5]; 5] = [[0; 5]; 5];

    println!("Enter: ");
    for i in 0..5 {
        for j in 0..5 {
            input.clear();
            std::io::stdin().read_line(&mut input).expect("Error");
            matrix[i][j] = input.trim().parse().expect("Error");
        }
    }

    let max_primes: usize = 0;
    let counter: usize = 0;

    basic_cycle(&matrix, max_primes, counter);

    if max_primes == 0 {
        println!("0");
    }
    else {
        println!("{}", counter);
        cool_cycle(&matrix, max_primes, counter);
    }
}
